### PR TITLE
Ft user email verification

### PR DESCRIPTION
**Pull request by Kellswork**
#### What does this PR do?
Implement email verification
#### How can it be manually tested?
......

#### Are there test specs for this?
......

#### Link to the ticket on `Notion`
[email verification](https://www.notion.so/5a87fb8fa8e1476ab938ae315d474e6e?v=157e0b179b404a3cadd1e943f0d2be95&p=9693a4ffdd424f41876269df7cb50302)

#### Any background contest?
......

### CODE REVIEW

File: src/resources/user/user.route.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { Router } from 'express';
-import { addUserInfo, login } from './auth.controller';
+import { addUserInfo, login, verifyEmail } from './auth.controller';
 import { validateUser, validateLogin } from '../../middlewares/validateUser';
 // import validateUserAddress from '../../middlewares/validateUserAddress';
```
Comment (omenkish) [inline src/resources/user/user.route.js:4]: Let's do away with the comment here

File: src/resources/user/user.route.js [hunk 2]
```diff
@@ -8,4 +8,5 @@ export default router;
 
 router.post('/auth/register', validateUser, addUserInfo);
 router.post('/auth/login', validateLogin, login);
+router.patch('/auth/verifyEmail', verifyEmail);
 // router.post('/users/:id/address', validateUserAddress, addUserAddressInfo);
```
Comment (omenkish) [inline src/resources/user/user.route.js:12]: This comment too

File: src/services/email.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+          from: 'LysterGems <support@lystergems.com>', // sender address
+          to: user.email, // list of receivers
+        },
+        locals: {
+          fname: user.firstname,
+          lname: user.lastname,
+          token,
+        },
+      },
+      console.log('email', user, token),
```
Comment (omenkish) [inline src/services/email.js:52]: let's remove this also

File: src/services/email.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
+import nodemailer from 'nodemailer';
+import Email from 'email-templates';
+import path from 'path';
+// const nodemailer = require('nodemailer');
```
Comment (omenkish) [inline src/services/email.js:4]: These comments too

File: src/resources/user/auth.controller.js [hunk 5]
```diff
@@ -24,13 +28,15 @@ export const addUserInfo = async (req, res) => {
       },
       token,
     };
+    sendEmailConfirmation(data.user, token);
```
Comment (omenkish) [inline src/resources/user/auth.controller.js:31]: You have access to the user object here without going through the data object

File: src/resources/user/auth.controller.js [hunk 6]
```diff
@@ -24,13 +28,15 @@ export const addUserInfo = async (req, res) => {
...
     return formatResponse(
       res,
       { message: 'user created successfully' },
       201,
       data,
     );
   } catch (error) {
+    console.log(error);
```
[the remark this answers is not in the document]
Comment (Kellswork) [reply src/resources/user/auth.controller.js:39]: sure

File: src/resources/user/auth.controller.js [hunk 7]
```diff
@@ -79,5 +85,36 @@ export const login = async (req, res) => {
     );
   }
 };
+export const verifyEmail = async (req, res) => {
+  try {
+    const { token } = req.query;
+    const validateToken = verifyToken(token);
+    // if the token has expired
```
[the remark this answers is not in the document]
Comment (Kellswork) [reply src/resources/user/auth.controller.js:92]: Nope, I was trying to remember something

File: src/resources/user/models/index.model.js [hunk 8]
```diff
@@ -12,3 +12,6 @@ export async function createUserAddress(userAddressInfo) {
 export async function getUserByEmail(email) {
   return User.query().where({ email });
 }
+export async function updateUserVerification(id) {
```
[the remark this answers is not in the document]
Comment (Kellswork) [reply src/resources/user/models/index.model.js:15]: okay lol I don't know why I'm replying

File: src/resources/user/auth.controller.js (hunk 5 above)
Comment (Kellswork) [reply src/resources/user/auth.controller.js:28]: If I destructure this, I change the format of how you pass data to the formatResponse() for user registration. something is making the test fail though it's either  me trying to destructure the data object or trying to make the expiresIn in jwt dynamic

Comment (omenkish) [reply src/resources/user/auth.controller.js:28]: you don't need to destructure, just do something like `sendEmailConfirmation(user, token);`. The user variable is in scope

Comment (Kellswork) [reply src/resources/user/auth.controller.js:28]: Okay yeah, let me try that

Comment (Kellswork) [reply src/resources/user/auth.controller.js]: Done

**Pull request merged**